test/integration/controller/jobframework/setup/setup_controllers_test.go

```diff
@@ -0,0 +1,134 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job
+
+import (
+	"errors"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	jobsetapi "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/jobset"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	testingjobset "sigs.k8s.io/kueue/pkg/util/testingjobs/jobset"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Setup Controllers", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	var (
+		ns           *corev1.Namespace
+		flavor       *kueue.ResourceFlavor
+		clusterQueue *kueue.ClusterQueue
+		localQueue   *kueue.LocalQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		fwk = &framework.Framework{CRDPath: crdPath}
+		cfg = fwk.Init()
+		ctx, k8sClient = fwk.SetupClient(cfg)
+		fwk.StartManager(ctx, cfg, managerSetup(jobframework.WithEnabledFrameworks([]string{jobset.FrameworkName})))
+
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "jobset-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		flavor = testing.MakeResourceFlavor("on-demand").Obj()
+		gomega.Expect(k8sClient.Create(ctx, flavor)).Should(gomega.Succeed())
+
+		clusterQueue = testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(
+				*testing.MakeFlavorQuotas(flavor.Name).
+					Resource(corev1.ResourceCPU, "5").
+					Obj(),
+			).Obj()
+
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+		localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, flavor, true)
+
+		fwk.StopManager(ctx)
+		fwk.Teardown()
+	})
+
+	ginkgo.It("Should setup controller and webhook after CRD installation", framework.SlowSpec, func() {
+		jobSet := testingjobset.MakeJobSet("jobset", ns.Name).
+			ReplicatedJobs(testingjobset.ReplicatedJobRequirements{
+				Name:        "replicated-job-1",
+				Replicas:    1,
+				Parallelism: 1,
+				Completions: 1,
+			}).
+			Suspend(false).
+			Queue(localQueue.Name).
+			Obj()
+
+		ginkgo.By("Check that the JobSet CRDs are not installed", func() {
+			gomega.Expect(errors.Is(k8sClient.Create(ctx, jobSet), &meta.NoKindMatchError{})).To(gomega.BeTrue())
```

Comment:
Instead of `To`, could we use the `gomega.BeComparableTo` and `cmpopts.EquoteErrors`?

Reply:
Done